Lib/idlelib/idle_test/mock_idle.py

```diff
@@ -29,6 +31,8 @@ def __call__(self, *args, **kwds):
         self.kwds = kwds
         if isinstance(self.result, BaseException):
             raise self.result
+        elif self.return_self:
```

Comment:
I am not sure what 'suppression' you are proposing, but I really want to be able to raise any exception.  Calltip tests should include one with eval mocked with Func(BaseException) to raise BaseException.  (see code).